### PR TITLE
Use filtersConf approach in Affected clusters table

### PR DESCRIPTION
**Pull request by ikerreyes**
Use the same criteria for the filters in AffectedClusterTable: use filtersConf and filtering utilities

### CODE REVIEW

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js [hunk 1]
```diff
@@ -57,11 +62,26 @@ data.forEach(
...
+  name: {
+    selectorText: 'Name',
+    values: SEARCH_ITEMS,
+    type: 'input',
+    filterFunc: (it, value) =>
+      it.name.toLowerCase().includes(value.toLowerCase()),
+  },
+  version: {
+    selectorText: 'Version',
+    values: VERSION_COMBINATIONS,
```
Comment (gkarat) [inline src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:75]: `VERSION_COMBINATIONS` contains nested arrays: https://github.com/RedHatInsights/ocp-advisor-frontend/blob/fa3793e788361d0cb2cd95962c5196257d24988f/cypress/utils/filters.js#L13. To get only the possible values, you have to flatten the array and remove duplicates

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js [hunk 2]
```diff
@@ -75,16 +95,20 @@ describe('test data', () => {
     cy.wrap(data).its('length').should('be.lte', 50);
   });
   it('has more than one enabled clusters with "custom" in name', () => {
-    cy.wrap(filterData('custom')).its('length').should('be.gt', 1);
+    cy.wrap(filterData({ name: 'custom' }))
+      .its('length')
+      .should('be.gt', 1);
```
Comment (gkarat) [inline src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:100]: Don't you want to use the `expect` notation here too already?

**Review by gkarat — COMMENTED** [review]
lint didn't pass:
```
ocp-advisor/src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js
  22:3  error  'TOOLBAR_FILTER' is defined but never used   no-unused-vars
  23:3  error  'DROPDOWN_TOGGLE' is defined but never used  no-unused-vars
  24:3  error  'DROPDOWN_ITEM' is defined but never used    no-unused-vars
  25:3  error  'ouiaId' is defined but never used           no-unused-vars

```

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js (hunk 1 above)
Comment (ikerreyes) [reply src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:75]: No, it is fine as it is. It is the expected behavior as it is a checkbox, so multiple values can be clicked. Have you seen something weird in the tests?

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js (hunk 2 above)
Comment (ikerreyes) [reply src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:100]: yes, but as it was part of another MR, I didn't want to merge concepts here. I will merge with the other and update everything.

File: src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js (hunk 1 above)
Comment (gkarat) [reply src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js:71]: @ikerreyes, oh, you are true, it works as expected. sorry for the noise

**Review by gkarat — APPROVED** [review]
looks good! 🦖

**Pull request merged**